pupil_src/shared_modules/calibration_choreography/base_plugin.py

```diff
@@ -123,6 +123,14 @@ class CalibrationChoreographyPlugin(Plugin):
     """
     is_session_persistent = True
 
+    @classmethod
+    def selection_label(cls) -> str:
+        return self.label
+
+    @classmethod
+    def selection_order(cls) -> float:
+        return float("inf")
+
```

Comment:
Why are these methods? They don't _do_ anything. Couldn't it just be a class variable?

Reply:
So I guess this change is technically possible
```suggestion
    selection_label: str = label
    selection_order: float = float("inf")
```
but the major difference would be the order of execution. If you want `selection_label` to fall back to the subclasses' `label` class attribute by default, then you need this to be a class method.

Reply:
Agreed, but we don't have a subclass without a `selection_label`. This looks like functionality we don't actually use.

Reply:
Also now that I look at it, the `label` in the base class is set to `None`, which isn't even a string. So it would actually break if you don't set a `label` in the subclass.

Reply:
Yes, you have to set a `label` in the subclass. But using the `selection_label` classmethod implementation above, you would not _have to_ overwrite the `selection_label`.

Reply:
Which is probably what we would be doing anyway if we are ever going to add another choreography. I just mean that this is extra code for a use case that we don't have and that will probably never happen.

Reply:
@pfaion from experience I found that using `@classmethod`s for things like this yields more flexible, understandable and predictable behaviour. You *could* do arbitrary transformations to produce the `selection_label` (including getting the label of the superclass, modifying it in some way, etc.). And since this is not performance critical code, I don't see any benefit switching to a class attribute. Ideally, this would be a `@classproperty` if existed, but since it doesn't, I think the flexibility of `@classmethod` is worth it.

Now that I'm looking at it, I think `label` was supposed to be a `@classmethod` as well, and instead of setting it to `None`, we would mark it as abstract for concrete subclasses to implement. If I remember correctly, that approach didn't work because we access `label` in `__init__subclass__`, but conceptually it is an abstract member for subclasses to implement.

Reply:
Ok.